tests/integration/docidcache/test_crud_cache.py

```diff
@@ -0,0 +1,215 @@
+    docs = list(get_documents(chunks=chunks, same_content=same_content, nr=nr))
+    assert len(docs) == nr
+    ids_used = set()
+    check_docs(chunk_content, chunks, same_content, docs, ids_used)
+
+    if nr > 0:
+        index_start = 1 + max(list(ids_used))
+    else:
+        index_start = 1
+    new_docs = list(get_documents(chunks=chunks, same_content=same_content, nr=nr, index_start=index_start))
+    new_ids = set([d.id for d in new_docs])
+    assert len(new_ids.intersection(ids_used)) == 0
+
+    check_docs(chunk_content, chunks, same_content, new_docs, ids_used, index_start)
+
+
+def check_docs(chunk_content, chunks, same_content, docs, ids_used, index_start=0):
+    for i, d in enumerate(docs):
+        i += index_start
+        id_int = int(d.id)
+        assert id_int not in ids_used
+        ids_used.add(id_int)
+
+        if same_content:
+            assert d.text == 'hello world'
+            np.testing.assert_almost_equal(d.embedding, d_embedding)
+        else:
+            assert d.text == f'hello world {i}'
+            assert d.embedding.shape == d_embedding.shape
+
+        assert len(d.chunks) == chunks
+
+        for j, c in enumerate(d.chunks):
+            id_int = int(c.id)
+            assert id_int not in ids_used
+            ids_used.add(id_int)
+            if same_content:
+                if chunk_content is None:
+                    chunk_content = c.content_hash
+                assert c.content_hash == chunk_content
+                assert c.text == 'hello world from chunk'
+                np.testing.assert_almost_equal(c.embedding, c_embedding)
+            else:
+                assert c.text == f'hello world from chunk {j}'
+                assert c.embedding.shape == c_embedding.shape
+
+
+@pytest.mark.parametrize('indexers, field, shards, chunks, same_content',
+                         [
+                             ('sequential', 'id', 1, 5, False),
+                             ('sequential', 'id', 3, 5, False),
+                             ('sequential', 'id', 3, 5, True),
+                             ('sequential', 'content_hash', 1, 0, False),
+                             ('sequential', 'content_hash', 1, 0, True),
+                             ('sequential', 'content_hash', 1, 5, False),
+                             ('sequential', 'content_hash', 1, 5, True),
+                             ('sequential', 'content_hash', 3, 5, True),
+                             ('parallel', 'id', 3, 5, False),
+                             ('parallel', 'id', 3, 5, True),
+                             ('parallel', 'content_hash', 3, 5, False),
+                             ('parallel', 'content_hash', 3, 5, True)
+                         ])
+def test_cache_crud(
+        tmp_path,
+        indexers,
+        field,
+        shards,
+        chunks,
+        same_content
+):
+    config(field, tmp_path, shards, indexers)
+    print(f'{tmp_path=}')
+    f = Flow.load_config('yml/flow.yml')
```

Comment:
can u use absolute path with reference filename dir so that test can be executed from different paths